examples/hello_world/src/main.rs

```diff
@@ -52,6 +57,36 @@ impl eframe::App for MyApp {
             ui.image(egui::include_image!(
                 "../../../crates/egui/assets/ferris.png"
             ));
+
+            let (rect, _) =
+                ui.allocate_exact_size(vec2(100.0, 50.0), Sense::focusable_noninteractive());
+
+            let stroke = PathStroke {
+                width: 1.5,
+                color: egui::epaint::ColorMode::UVBounds(
+                    rect,
+                    Arc::new(Box::new(|p: Pos2| {
+                        if p.x < 0.5 {
+                            Color32::RED
+                        } else {
+                            Color32::GREEN
+                        }
+                    })),
+                ),
+            };
+
+            ui.painter_at(rect).add(Shape::line_segment(
+                [rect.left_center(), rect.right_center()],
+                stroke.clone(),
+            ));
+
+            ui.painter_at(rect).add(Shape::line_segment(
+                [
+                    rect.left_center() + vec2(25.0, 15.0),
+                    rect.right_center() + vec2(0.0, 15.0),
+                ],
+                stroke.clone(),
+            ));
```

Comment:
This is not the right place for this test/example.

Either add a standalone example, or add a new demo to `egui_demo_lib`

Reply:
resolved in a8e0e57dc2a4e0cc2933dd3c8bfd146beb88d2de